src/reporter/dwcReporter.js

```diff
@@ -0,0 +1,308 @@
+const request = require("request");
+const uuid = require("uuid");
+
+const status = {
+  running: "running",
+  success: "successful",
+  failed: "failed"
+};
+
+
+/**
+ * @typedef DwCReporterConfig
+ * @type {Object}
+ * @extends {Config}
+ */
+
+/**
+ * @typedef DwCReporterInstanceConfig
+ * @type {Object}
+ */
+
+/**
+ * Report test execution
+ * @constructor
+ * @param {DwCReporterConfig} config
+ * @param {DwCReporterInstanceConfig} instanceConfig
+ * @param {Logger} logger
+ * @param {StatisticCollector} collector
+ */
+function DwcReporter(config,instanceConfig,logger,collector) {
+  this.config = config || {};
+  this.instanceConfig  = instanceConfig || {};
+  this.logger = logger;
+  this.collector = collector;
+  this.options = {
+    stdout: true,
+    syncInterval: 100,
+    retries: 5,
+    themistoUrl: instanceConfig.themistoUrl,
+    themistoCredentials: `Basic ${Buffer.from(`${instanceConfig.themistoUser}:${instanceConfig.themistoPass}`).toString("base64")}`,
+    vector: instanceConfig.vector
+  };
+}
+
+// upload data in Themisto
+DwcReporter.prototype._postMetadata = function(url, credentials, vectorId, value) {
```

Comment:
why do you need to give this url as param to every function, it is always this.thermostroUrl ?